modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/NotificationsSettingsFragment.kt

```diff
@@ -445,6 +399,49 @@ class NotificationsSettingsFragment :
         }
     }
 
+    private fun setupEnabledNotifications() {
+        launch(Dispatchers.Default) {
+            val enabled = settings.notifyRefreshPodcast.flow.value
+
+            launch(Dispatchers.Main) {
+                enabledPreference?.isChecked = enabled
+                enabledPreferences(enabled)
+
+                enabledPreference?.setOnPreferenceChangeListener { _, newValue ->
+                    val checked = newValue as Boolean
+                    settings.notifyRefreshPodcast.set(checked)
+
+                    analyticsTracker.track(
+                        AnalyticsEvent.SETTINGS_NOTIFICATIONS_NEW_EPISODES_TOGGLED,
+                        mapOf("enabled" to checked)
+                    )
+
+                    lifecycleScope.launch {
```

Comment:
Interestingly, if I used the Fragment's CouroutineScope directly here by calling `launch` instead of `lifecycleScope.launch` the coroutine wasn't getting run. I'm not sure why that is. 😕 If you have any insight, let me know.